src/Fixer/ClassNotation/OrderedClassElementsFixer.php

```diff
@@ -0,0 +1,399 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\ClassNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\ConfigurationException\InvalidFixerConfigurationException;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Gregor Harlan <gharlan@web.de>
+ */
+final class OrderedClassElementsFixer extends AbstractFixer
+{
+    /**
+     * @var array Array containing all possible configuration types (keys) and their parent types (values)
+     */
+    private static $typeHierarchy = array(
+        'use_trait' => null,
+        'public' => null,
+        'protected' => null,
+        'private' => null,
+        'constant' => null,
+        'constant_public' => array('constant', 'public'),
+        'constant_protected' => array('constant', 'protected'),
+        'constant_private' => array('constant', 'private'),
+        'property' => null,
+        'property_static' => array('property'),
+        'property_public' => array('property', 'public'),
+        'property_protected' => array('property', 'protected'),
+        'property_private' => array('property', 'private'),
+        'property_public_static' => array('property_static', 'property_public'),
+        'property_protected_static' => array('property_static', 'property_protected'),
+        'property_private_static' => array('property_static', 'property_private'),
+        'method' => null,
+        'method_static' => array('method'),
+        'method_public' => array('method', 'public'),
+        'method_protected' => array('method', 'protected'),
+        'method_private' => array('method', 'private'),
+        'method_public_static' => array('method_static', 'method_public'),
+        'method_protected_static' => array('method_static', 'method_protected'),
+        'method_private_static' => array('method_static', 'method_private'),
+        'construct' => null,
+        'phpunit' => null,
+    );
+
+    /**
+     * @var string[] Default order/configuration
+     */
+    private static $defaultOrder = array(
+        'use_trait',
+        'constant_public',
+        'constant_protected',
+        'constant_private',
+        'property_public',
+        'property_protected',
+        'property_private',
+        'phpunit',
+        'construct',
```

Comment:
Actually I can't find any example in code.

> Declare public methods first, then protected ones and finally private ones. The exceptions to this rule are the class constructor and the setUp and tearDown methods of PHPUnit tests, which should always be the first methods to increase readability;

I just took order of elements in sentence. Nothing more strict.